### PR TITLE
Add route entity & update generateClientCommand

### LINKED ISSUE #154: Add add route feature
Dear All,
這邊先放我 trace code 的結果，整理一下我要做的事情。

## 流程
[CreatePod](https://github.com/linkernetworks/vortex/blob/develop/src/pod/pod.go#L199) -> [generateNetwork](https://github.com/linkernetworks/vortex/blob/develop/src/pod/pod.go#L158) -> [generateInitContainer](https://github.com/linkernetworks/vortex/blob/develop/src/pod/pod.go#L108) -> **[generateClientCommand](https://github.com/linkernetworks/vortex/blob/develop/src/pod/pod.go#L94)** -> **[Network-controller client](https://github.com/linkernetworks/network-controller/blob/master/client/main.go)** -> Network-controller server 建 pod 的網路設定包含(config interface, add route)

相關 Entity: **[PodNetwork](https://github.com/linkernetworks/vortex/blob/develop/src/entity/pod.go#L19)**

粗體的是要加 code 的地方

----

## 文件調整
https://github.com/linkernetworks/vortex/blob/develop/API.md#create-pod

---
## client option
- network-controller client 需要額外傳給 network-controller server 的參數：
   - [dstCIDR](https://github.com/linkernetworks/network-controller/blob/master/messages/messages.proto#L165): 一定要餵值，但要判斷值正不正確。string、要符合 CIDR 格式(network-controller server 已經完成)。
   - [gwIP](https://github.com/linkernetworks/network-controller/blob/master/messages/messages.proto#L166): 可以空值
- 設計
   - 第一版：一個指令、一個 interface、一個 route ("`-r 239.0.0.0/4 -gw 0.0.0.0`" or "`-r 239.0.0.0/4`")

---

## Todolist
- [x] Write [Network-controller client](https://github.com/linkernetworks/network-controller/blob/master/client/main.go) - linkernetworks/network-controller#65 📅  7/24 Done
使用 `--net 239.0.0.0/4 -g 0.0.0.0` or `--net 239.0.0.0/4 --gateway 0.0.0.0`。目前一行指令中可設定一個 interface 可以加一個 IP routing table。
- [x] Update [PodNetwork entity](https://github.com/linkernetworks/vortex/blob/develop/src/entity/pod.go#L19)
- [x] Write [generateClientCommand](https://github.com/linkernetworks/vortex/blob/develop/src/pod/pod.go#L92)

### PR DESCRIPTION
**Pull request by sufuf3**
Related issue: #154 
close #154 

- Update PodNetwork Entity: Add PodRoute structure & add a routes attribute.
- Fix typo of comment
- Update "generateClientCommand"  to support add route command. In this time, I just use array[0], because it's limited with Network-controller client the main function.

### CODE REVIEW

File: src/entity/pod.go [hunk 1]
```diff
@@ -18,17 +18,24 @@ type Container struct {
 	Command []string `bson:"command" json:"command" validate:"required,dive,required"`
 }
 
+// PodRoute is the structure for add IP routing table
+type PodRoute struct {
+	DstCIDR string `bson:"dstCIDR" json:"dstCIDR" validate:"required,cidr"`
+	Gateway string `bson:"gateway" json:"gateway" validate:"min=1,max=4095"`
```
Comment (hwchiu) [inline src/entity/pod.go]: min,max validate for string gateway ?

Comment (sufuf3) [reply src/entity/pod.go]: Sorry, My bad. Fix soon.

Comment (John-Lin) [reply src/entity/pod.go]: max=4095,min=0

Comment (sufuf3) [reply src/entity/pod.go]: Vlan part, update here linkernetworks/network-controller#67

Gateway use ip validate, this part was updated in this PR.

**Comment by sufuf3** [comment]
Dear All, please help me review this PR.
Thank you.

**Comment by sufuf3** [comment]
Rebase to latest develop branch. Done

**Comment by sufuf3** [comment]
I have created a new branch(used the latest develop branch) & used format-patch & am do for this PR.

```sh
* 3392ab3 Add add route in generateClientCommand, cc #154 sufuf3 2 hours ago
* a1f1459 Fix typo of PodVolume struct comment sufuf3 4 hours ago
* d3b7516 Add route entity, cc #154 sufuf3 4 hours ago
*   bd21f12 Merge pull request #172 from linkernetworks/phstsai/VX-214 Che-Wei Lin 16 minutes ago
|\
| * d9486c6 improve unit test for query situation Pei Hsuan Tsai 3 hours ago
|/
* b0b2372 Change the response from bad-request to method not allowed hwchiu 5 hours ago
* deebf06 Use the kubeutil to check the pod status when deleteing the volume hwchiu 26 hours ago
* dc33188 Return the badrequest if the network still used by some Pods when delete the network hwchiu 26 hours ago
* 402f454 remove race John-Lin 3 hours ago
```

**Pull request merged**